app/services/v1/chart_statistics/create.rb

```diff
@@ -72,7 +72,7 @@ def zero_division_error?
 
   def inside_date_range?
     return false if chart.date_range_start.present? && chart.date_range_start > user_session.finished_at
-    return false if chart.date_range_end.present? && chart.date_range_end < user_session.finished_at
+    return false if chart.date_range_end.present? && chart.date_range_end + 1.day <= user_session.finished_at
```

Comment:
FE sends the *beginning* of the last day in the date range, so we need to offset it by 1

Reply:
so why not chart.date_range_end.end_of_day?

Reply:
I think it is more intuitive

Reply:
The timestamp FE sends doesn't include the timezone, and I have agreed with @bartosz-kepka-htd that's going to be the best solution